manageiq-operator/Dockerfile

```diff
@@ -1,5 +1,10 @@
 # Build the manager binary
-FROM docker.io/library/golang:1.23 as builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal as builder
+
+RUN microdnf -y install gzip tar && \
+    curl -f -L https://go.dev/dl/go1.23.1.linux-amd64.tar.gz | tar -C / -xzf -
```

Comment:
This has to be made arch aware for multi-arch builds right?

Reply:
The version number is gonna be a pain.  For example 1.23.2 is already out so this PR is out of date.

Reply:
Done